packages/app/src/specs/generators/EmptyGenerator.vue

```diff
@@ -190,6 +195,24 @@ const hasError = computed(() => !isValidSpecFile.value && !!specFile.value)
 
 const result = ref<GeneratorSuccessFileFragment | null>(null)
 
+onMounted(() => {
+  // Focus text input and try to pre-select file name
+  inputRef.value?.focus()
+
+  // This regex matches a group of word characters, spaces, and '-' characters before a '.' character.
```

Comment:
As a nit/discussion item but not a hard blocker, I want to get your opinion on how we should be testing this kind of code, specifically the "business logic" part, which is parsing the spec.

Eg, I wonder if a pure function like this would be better:

```js
const fileNameRegex = /[ \w-]+(?=\.)/

export function getHighlightRange (filename: string) {
  const match = inputValue?.match(fileNameRegex)

  if (!match) {
    return null
  }

  const startSelectionIndex = match.index || 0
  const endSelectionIndex = startSelectionIndex + match[0].length
   
  return [startSelectionIndex, endSelectionIndex]
}
```

It would be easy to test in isolation (no need for the UI framework to come into play), and against lots of different strings - right now, we only test it against one string.

Finally, you'd the import this into the component, and your `onMount` would be

```js
onMounted(() => {
  if (!inputRef?.value) {
    return
  }

  const match = getSelectionRange(specFile)
  
  if (!match) {
    return
  }

  inputRef.value.setSelectionRange(match[0], match[1])
})
```

More a point of discussion around best practices and testing philosophy in general, thoughts? It's a bit more overhead, but I like the separation of concerns.

Reply:
Yeah this is a good discussion to have. I like the idea of having at least one component test that tests the "happy path" just so that we are sure that in most cases the component is behaving as expected. Having at least one test that covers everything together is crucial imo. But I also have this same thought

> I'm not sure the ideal of re-mounting a component many times for lots of regexp test is ideal.

One thing I'd question about that though: is the regex the _only_ thing we'd be testing? I'd argue that it's not. What's important is the experience for the user for different situations (i.e. when the default file path that we provide doesn't match the regex) which I think we can get better from a CT.

So at this point my thought is to add more component test cases that provide different default strings and see how the component behaves. That's probably what I'll do to finish out this PR, but I'm not sure if I love that approach because it doesn't seem very efficient.

I also like the idea of separating the logic out into its own function like `getSelectionRange` for the sake of cleaning things up a bit if nothing else. As far as using CT vs a normal unit test for something like this, I think I'm torn. Would like to hear what the rest of the team thinks

Reply:
Good points! No right or wrong here.

For now, happy with the unhappy case you've added: https://github.com/cypress-io/cypress/pull/22284/commits/50851bd064854d9f42ff87c2c26e782a2b678c5c

✅